Web/Teleterm: Update UI health warning indicators

GavinFrazar opened the pull request:
This changes fetches all health statuses for HA deployed resources that
are health checked (databases for now) and shows warning indicators if
any are unhealthy or a mix of healthy/unknown/unhealthy (likely
misconfiguration).


Part of:
- #20544

Stacked on top of another PR:
- https://github.com/gravitational/teleport/pull/54868

kimlisa commented on web/packages/teleport/src/config.ts:
```diff
@@ -279,7 +279,7 @@ const cfg = {
     databasesPath: `/v1/webapi/sites/:clusterId/databases?searchAsRoles=:searchAsRoles?&limit=:limit?&startKey=:startKey?&query=:query?&search=:search?&sort=:sort?`,
 
-    databaseServers: {
+    databaseServer: {
```
i reverted the changes to this, b/c this just describes the resource (like `db_server`) and it contains different actions against the resource like [this](https://github.com/gravitational/teleport/blob/5b686f538281679175cdd231178c85018adb9472/web/packages/teleport/src/config.ts#L911)

kimlisa commented on web/packages/shared/components/UnifiedResources/shared/StatusInfo.story.tsx:
```diff
@@ -37,7 +36,7 @@ import {
   attemptState: 'success' | 'processing' | 'failed' | '';
   resourceKind: 'db';
-  healthStatus: ResourceStatus | 'empty';
+  healthStatus: 'unhealthy' | 'unknown' | 'unhealthy-unknown';
```
it determines the health status of the servers listed in the story. i added a story comment to clarify

kimlisa commented on lib/teleterm/apiserver/handler/handler_unified_resources.go:
```diff
@@ -52,6 +52,7 @@ func (s *Handler) ListUnifiedResources(ctx context.Context, req *api.ListUnified
 		UseSearchAsRoles:    req.GetSearchAsRoles(),
 		PinnedOnly:          req.GetPinnedOnly(),
 		IncludeRequestable:  req.GetIncludeRequestable(),
+		IncludeHealthStatus: true,
```
i'm actually unsure about this.

this looks like a flag that should be controlled by the caller... i think we do use `ListResources` elsewhere that may not need the health status. does including health status slow things down?

what's everyone opinion?

kimlisa committed:
Web/Teleterm: Update UI health warning indicators

The UI will now show the warning indicator if the health status is
either "unhealthy" or "mixed".
The status panel will now show the health statuses of all servers for
the resource.
